internal/internal_task_handlers.go

```diff
@@ -460,6 +479,18 @@ func (w *workflowExecutionContextImpl) Unlock(err error) {
 	w.mutex.Unlock()
 }
 
+func (w *workflowExecutionContextImpl) getEventHandler() *workflowExecutionEventHandlerImpl {
+	eventHandler := w.eventHandler.Load()
```

Comment:
Why mutex is not enough for race condition?

Reply:
Synced offline on this. I agree that we are misusing the lock. But fixing that requires rewriting lots of task processing logic and is a long term plan. This pr just a temp bug fix for preventing our customer running into this situation again.